core/services/pipeline/orm.go

```diff
@@ -458,17 +458,20 @@ func (o *orm) ResultsForRun(ctx context.Context, runID int64) ([]Result, error)
 }
 
 func (o *orm) RunFinished(runID int64) (bool, error) {
-	var done struct{ Done bool }
 	// TODO: Since we denormalised this can be made more efficient
 	// https://www.pivotaltracker.com/story/show/176557536
+	var tr TaskRun
 	err := o.db.Raw(`
-        SELECT finished_at IS NOT NULL AS done
+        SELECT * 
         FROM pipeline_task_runs
         INNER JOIN pipeline_task_specs ON pipeline_task_runs.pipeline_task_spec_id = pipeline_task_specs.id
         WHERE pipeline_task_runs.pipeline_run_id = ? AND pipeline_task_specs.successor_id IS NULL
 		LIMIT 1
-    `, runID).Scan(&done).Error
-	return done.Done, errors.Wrapf(err, "could not determine if run is finished (run ID: %v)", runID)
+    `, runID).Scan(&tr).Error
```

Comment:
Scan does not error if nothing returned

Reply:
Shouldn't you check `err` first here?

Reply:
This query can be simplified anyway, check the ticket

Reply:
this code isn't used at the moment and is possible we may not bring it back - will hold off on doing the TODO until we know the plan for this v2 non-optimized path